osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/services/SecretManagementService.java

```diff
@@ -247,7 +252,70 @@ private EncryptedTypedSecret retrieveSecret(
     return Optional.of(secretsList.iterator().next());
   }
 
-  public synchronized void storeSecrets(
+  public synchronized void storeSingleNewSecrets(
```

Comment:
Just saying: this method is synchronized, but we use multiple instances of secret-management. So it does not mean that this is the only place where this code is executed at the same time! Someone should look into this!

Reply:
I'm not sure if the function name `storeSingleNewSecrets` is clear. `Single` in combination with a list as parameter is confusing. I would expect something like `validateAndStoreNewSecrets`.

Reply:
Method name is confusing indeed. I suggest storeOrResetNewSecrets. That is exactly what is does.

Reply:
After rework the method is named storeSecrets as it initialy was

Reply:
In protocol-adapter-dlms a mechanism is introduced to ensure key changing processes to be never executed in parallel. storeSecrets methosd is used from this dlms-adapter. I'm not sure about calls from other modules.

Reply:
See comment above